content/projects/tdd/simple-calculator-part1/_index.md

```diff
@@ -4,33 +4,88 @@ pre: "<b>EASY: </b>"
 
-Also, please don't build a frontend.
+The basic TDD approach is as follows:
 
-We'll follow a TDD approach. The basic approach is
+1. RED: Write tests. It should fail initially because there isn't any code that it is testing.
+2. GREEN: Write code to make the tests pass
+3. REFACTOR: Make sure code is understandable and clean.
 
-1. RED: we will write ONE failing test
-2. GREEN: we will make sure our test (and all tests written so far) pass
-3. REFACTOR: if our code is untidy we will clean it up
-4. repeat all the steps until all the functionality we need has been built and tested
+Remember to make sure your tests still pass after refactoring it.
 
-## 1. Make a function that adds two numbers
+## Set up environment
 
-First write a test function. it can be called `test_function` or something similar. It needs to make sure that the `add` function works. (of course we haven't written `add` yet. We'll write our tests first).
+### JavaScript:
 
-- write a test to demonstrate that `add(0,0)` will return 0. Run the test. It should fail. Now make the test pass by creating an `add` function.
-- write a test to demonstrate that `add(-1,-1)` should return -2 then edit `add` to make your test pass
-- write a test to demonstrate that `add(4,5)` should return 9 then edit `add` to make the test pass
+Use Jasmine to test your code. *Please do not use the SpecRunner* to test your code. Run Jasmine on the terminal.
+- [Running Jasmine on the Terminal](../../../topics/jasmine-unit-tests/_index.md). Look under the heading: *Getting set up (like a boss)* for instructions to set up.
 
-## 2. Make sure you can add as many numbers as you want
+After setting up Jasmine on the terminal, please ensure that your directory has the following:
 
-Now write a test to make sure you can add a bunch of numbers. Eg `add(1,2,3,4)` should return 10. Think of the tests yourself
+- A `src` folder with your source file : `simpleCalculator.js` .
+- A `spec` folder with your test file: `simple-Calc-part1_spec.js`.
 
-## 3. Create a multiply function that can multiply 2 numbers
+Your directory structure should look like this:
 
-`multiply(1,2)` and other similar combinations should work. Run with it
+```
+>simple-calculator-part1
+    >node_modules
+    >spec   
+        > support
+            -jasmine.json
+        - simple-Calc-part1_spec.js
+    >src
+        -simpleCalculator.js
+    -package.json
+```
 
-## 4. Make sure multiply can handle multiple numbers
+## 1. Create an add function that can add two integers
+
+Implement the following function:
+```
+add()
+```
+
+The `add` function should behave like this:
+
+```
+add(1,2)
+// should return 3
+add(-1,-1)
+// should return -2
+```
+
+## 2. Modify the add function so that it can add multiple integers.
+
+The `add` function should now behave like this:
+
+```
+add(1,2,3,4,5)
```

Comment:
One common mistake people make is: They break their add function so that it no longer works with 2 numbers.  So it would be good to be very explicit here:

```
add(1,2,3,4)
// should return 10
add(1,2,3,4,5)
// should return 15
add(1,2)
// should still return 3
add(-1,-1)
// should still return -2
```

Reply:
Add a note: Also please note implementing a function that expects an array or a list as the input is wrong

```
add([1,2,3,4])   // this is NOT what we are looking for. If you have square brackets inside your round brackets, you are doing it wrong
```